### PR TITLE
Add base website

### PR DESCRIPTION
**Pull request by wtrocki**
Hi

When promoting reference architecture I got some feedback that generally documentation is hard to read in github.
This website is using Docusaurus. 
I have done some very quick prototype of the website that is currently hosted on my fork:

## How this looks like

https://wtrocki.github.io/nodejs-reference-architecture/

## Notes

Changes here are not relevant that much as I have been doing some copy paste, using some shortcuts to get things render easily

## Styling 

Order of chapter is random (possibly needs better work)

I did not put any effort into styling website but we can easily make it look amazing like this:

- https://offix.dev 
- https://graphback.dev

### CODE REVIEW

**Comment by wtrocki** [comment]
FYI @nodeshift/reference-architecture ?

**Comment by BethGriggs** [comment]
+1 to hosting the content in a more readable website form like this. I think it would be good to get it hosted under the  https://nodeshift.dev/ domain (we can figure out how to align the styling in due course).

**Comment by wtrocki** [comment]
@BethGriggs Thanks. Domain/subdomain is quite fast setup with docusaurus. So it can be either path or subdomain.
https://spec.nodeshift.dev etc.

I personally tend to actually register domains that reflect specifics of the website its serves.
For example 
https://nodeshift-spec.org

File: website/src/components/HomepageFeatures.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        'Enteprise Ready
+      </>
+    ),
+  },
+  {
+    title: 'Used in production',
+    Svg: require('../../static/img/undraw_docusaurus_tree.svg').default,
+    description: (
+      <>
+       Set of packages used in production by Red Hat and IBM teams that guarantee support and smooth experience
```
Comment (mhdawson) [inline website/src/components/HomepageFeatures.js:20]:
```
Based on experience from use within Red Hat and IBM teams in production and with customers.
```

File: website/src/components/HomepageFeatures.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      </>
+    ),
+  },
+  {
+    title: 'Community Driven',
+    Svg: require('../../static/img/undraw_docusaurus_react.svg').default,
+    description: (
+      <>
+        Build with collaboration with various communities
```
Comment (mhdawson) [inline website/src/components/HomepageFeatures.js:29]:
```suggestion
        Built with collaboration with various teams across Red Hat and IBM.
```

File: website/docusaurus.config.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/** @type {import('@docusaurus/types').DocusaurusConfig} */
+module.exports = {
+  title: 'Node.JS Reference Specification',
```
Comment (mhdawson) [inline website/docusaurus.config.js:3]:
```suggestion
  title: 'Node.JS Reference Architecture',
```

File: website/docusaurus.config.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  url: 'https://wtrocki.github.io',
+  baseUrl: '/nodejs-reference-architecture/',
+  onBrokenLinks: 'ignore',
+  onBrokenMarkdownLinks: 'warn',
+  favicon: 'img/favicon.ico',
+  organizationName: 'wtrocki', // Usually your GitHub org/user name.
+  projectName: 'nodejs-reference-architecture', // Usually your repo name.
+  themeConfig: {
+    navbar: {
+      title: 'Node.JS Reference Specification',
```
Comment (mhdawson) [inline website/docusaurus.config.js:14]:
```suggestion
      title: 'Node.JS Reference Architecture',
```

File: website/docusaurus.config.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      logo: {
+        alt: 'Node.JS Reference Specification Logo',
```
Comment (mhdawson) [inline website/docusaurus.config.js:16]:
```suggestion
        alt: 'Node.JS Reference Architecture Logo',
```

File: website/docusaurus.config.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      ],
+    },
+    footer: {
+      style: 'dark',
+      links: [
+        {
+          title: 'Docs',
+          items: [
+            {
+              label: 'Reference Specification',
```
Comment (mhdawson) [inline website/docusaurus.config.js:40]:
```suggestion
              label: 'Reference Architecture',
```

File: website/src/components/HomepageFeatures.js (hunk 1 above)
Comment (wtrocki) [reply website/src/components/HomepageFeatures.js:20]: I did not put much effort into actual messages/styling, before knowing what is your overal take on this. If we like idea we can do proper design/text

**Comment by helio-frota** [comment]
+1 for website

**Comment by lholmquist** [comment]
>  don't think we can use the Node.js logo unfortunately. @lholmquist would you have any suggestions of what to use instead?

Nodeshift logo could work.  I believe i have a link to all the various sizes

**Comment by wtrocki** [comment]
https://design.jboss.org/

**Comment by wtrocki** [comment]
Update from call:
- Docusaurus us good choice for now
- we will keep it minimal
- we need cicd job to auto update it
- @mhdawson to check with support to setup nodeshift subdomain.

Really glad we got this agreement sorted so fast. Thank you so much node.js team

**Comment by mhdawson** [comment]
In terms of

> @mhdawson to check with support to setup nodeshift subdomain pointing to github pages

I'll do this after we have the site up and published in github pages.

**Comment by wtrocki** [comment]
Good. Closing as it outdated. I will do this again

**Comment by wtrocki** [comment]
This got seriously outdated. Created new one

**Pull request closed without merging**